File: mooc_application/src/main/java/com/hongxuan/mooc/service/ProfessionalService.java
package com.hongxuan.mooc.service;
import com.hongxuan.pojo.TbProfessional;
import entity.PageResult;

import java.util.List;
/**
 * 业务逻辑接口
 * @author LiaoHongXuan
 *
 */
public interface ProfessionalService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbProfessional> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbProfessional professional);
	
	
	/**
	 * 修改
	 */
	public void update(TbProfessional professional);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbProfessional findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbProfessional professional, int pageNum,int pageSize);
	
}

File: mooc_application/src/main/webapp/js/service/professionalService.js
//服务层
app.service('professionalService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../professional/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../professional/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../professional/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../professional/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../professional/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../professional/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../professional/search.do?page='+page+"&rows="+rows, searchEntity);
	}    	
});
